x/auction/keeper/msg_server.go

```diff
@@ -17,10 +16,9 @@ var _ types.MsgServer = Keeper{}
 func (k Keeper) SubmitBid(c context.Context, msg *types.MsgSubmitBidRequest) (*types.MsgSubmitBidResponse, error) {
 	ctx := sdk.UnwrapSDKContext(c)
 
-	// Verify signer is the same as the bidder (this validates both the bidder and signer addresses)
-	signer := msg.MustGetSigner()
-	if !signer.Equals(sdk.AccAddress(msg.Bidder)) {
-		return &types.MsgSubmitBidResponse{}, sdkerrors.Wrapf(types.ErrSignerDifferentFromBidder, "Signer: %s, Bidder: %s", msg.GetSigner(), msg.GetBidder())
+	bidder, err := sdk.AccAddressFromBech32(msg.Bidder)
```

Comment:
If we're dropping `MustGetSigner` and using the bidder for `GetSigners`, we should retrieve it using `MustGetBidder` here. Although I would consider keeping it as `MustGetSigner` and just using the bidder as the signer field.

Reply:
sure that's fair, will update.

Reply:
Updated.